puppet-client.html

```diff
@@ -333,35 +328,48 @@
                                 break;
                             case 'replace':
-                                templateDomBind.splice(parentsPolymerPath, parseInt(name, 10), 1, operation.value);
+                                    templateDomBind.notifySplices(parentsPolymerPath,
+                                            [{
+                                                index: parseInt(name, 10),
+                                                removed: [result.removed],
+                                                addedCount: 1,
+                                                object: parent,
+                                                type: 'splice' }]
+                                    );
                                 break;
-                                // addedCount = 1;
-                                // var col = Polymer.Collection.get(parent);
-                                // removed = col.getItem('#'+name);
                             case 'remove':
-                                templateDomBind.splice(parentsPolymerPath, parseInt(name, 10), 1);
-                                //console.info('Remove is fragile, as Polymer requires something they call normalization, so splices to appear in index order, what is not guaranteed by this element.')
+                                templateDomBind.notifySplices(parentsPolymerPath,
+                                        [{
+                                            index: parseInt(name, 10),
+                                            removed: [result.removed],
+                                            addedCount: 0,
+                                            object: parent,
+                                            type: 'splice' }]
+                                );
                                 break;
                         }
                     } else {
                         switch (operation.op) {
                             case 'add':
                             case 'replace':
-                                templateDomBind.set(polymerPath, operation.value);
-                                // this.notifyPath(polymerPath, operation.value);
+                                templateDomBind.notifyPath(polymerPath, operation.value, false);
                                 break;
                             case 'remove':
-                                templateDomBind.set(polymerPath, null);
-                                templateDomBind.set(polymerPath, undefined);
-                                // this.notifyPath(polymerPath, undefined);
                                 //console.warn('Polymer does not support unsetting properties https://github.com/Polymer/polymer/issues/2565');
                                 //console.warn('remove is translated to JSON incompatible set to `undefined`, as there is no remove method in Polymer.')
+                                templateDomBind.notifyPath(polymerPath, undefined, false);
```

Comment:
It probably has the same problems. I just took suggestion from the comment (L362) :D